### PR TITLE
chore: improve readme and docker setup

### PR DESCRIPTION
**Pull request by ooliver1**
This:
- adds all the required vars in `.env.example`
- adds the internal network to docker-compose to connect to consul
- adds the required deps to poetry including formatters
- actually uses poetry in docker
- explains how to run much better

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -7,4 +7,7 @@ GUILD_ID=...                        # The guild id the bot uses
 BOT_LINKING_LOG_CHANNEL_ID=...      # Logs for actions by the bot linking cog
 BOOSTER_ROLE_ID=...                 # The booster role. Currently used for booster bots
 STARS_CHANNEL_ID=...                # The voice channel with the GitHub star count
-AUTO_THREAD_CHANNEL_ID=...          # Channel in which all messages are automatically turned into threads
\ No newline at end of file
+AUTO_THREAD_CHANNEL_ID=...          # Channel in which all messages are automatically turned into threads
+HEL_BANNED_ROLE_ID=...              # THe role that is not allowed to interact with the help channel
```
Comment (Skelmis) [inline .env.example:11]:
```python
HELP_BANNED_ROLE_ID=...              # The role that is not allowed to interact with the help channel
```

**Review by Skelmis — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -1,8 +1,14 @@
-FROM python:3.9
-WORKDIR opt/
-RUN pip install jishaku
-RUN pip uninstall discord.py
-RUN pip install algoliasearch
-RUN pip install git+https://github.com/nextcord/nextcord.git
-COPY . ./
-CMD ["python", "main.py"]
+FROM --platform=amd64 python:3.10-slim-buster
```
Comment (TAG-Epic) [inline Dockerfile:1]: Why specify the arch?

File: docker-compose.yml [hunk 3]
```diff
@@ -6,3 +6,6 @@ services:
             dockerfile: Dockerfile
         env_file:
             - .env
+
+        extra_hosts:
+            - "host.docker.internal:host-gateway"
```
Comment (TAG-Epic) [inline docker-compose.yml:11]: What does this do? Shouldnt this also contain consul?

**Review by TAG-Epic — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 3 above)
Comment (ooliver1) [reply docker-compose.yml:11]: host.docker.internal would link to the host, I wasn't too sure how to link consul from the host into the container

Comment (TAG-Epic) [reply docker-compose.yml:11]: Why not make a consul container inside the compose file?

**Pull request merged**